src/models/db/accounts.js

```diff
@@ -1,36 +1,121 @@
+  data.is_admin = Boolean(data.is_admin)
+  // oracle has balance stored properly, but knex returns it with a small
+  // rounding error, possibly a bug in knex; using toFixed as workaround
+  data.balance = Number(Number(data.balance).toFixed(config.amount.scale))
+  if (data.minimum_allowed_balance === null) {
+    data.minimum_allowed_balance = Number.NEGATIVE_INFINITY
+  } else if (data.minimum_allowed_balance) {
+    data.minimum_allowed_balance = Number(data.minimum_allowed_balance)
+  } else {
+    data.minimum_allowed_balance = 0
+  }
+  delete data.created_at
+  delete data.updated_at
+  return data
+}
+
+function convertToPersistent (data) {
+  data = _.cloneDeep(data)
+  if (data.balance) {
+    data.balance = Number(Number(data.balance).toFixed(config.amount.scale))
+  }
+  if (data.minimum_allowed_balance) {
+    if (data.minimum_allowed_balance === Number.NEGATIVE_INFINITY) {
+      data.minimum_allowed_balance = null
+    } else {
+      data.minimum_allowed_balance = Number(data.minimum_allowed_balance)
+    }
+  }
+  return _.transform(data, (result, value, key) => {
+    result[key.toUpperCase()] = value
+  })
+}
+
+function getTransaction (options) {
+  return !options ? knex : (!options.transaction ? knex : options.transaction)
+}
 
 function * getAccounts (options) {
-  const jsonAccounts = yield Account.findAll(options)
-  return jsonAccounts.map(Account.fromDatabaseModel.bind(Account))
+  return getTransaction(options)
+    .from(TABLE_NAME).select()
+    .then((results) => results.map(convertFromPersistent))
 }
 
 function * getConnectorAccounts (options) {
-  const jsonAccounts = yield Account.findAll(options)
-  return jsonAccounts.filter((account) => account.connector)
-    .map(Account.fromDatabaseModel.bind(Account))
+  return (yield getAccounts(options))
+    .filter((account) => account.connector)
+    .map(convertFromPersistent)
+}
+
+function getAccountByKey (key, value, options) {
+  return getTransaction(options)
+    .from(TABLE_NAME).select().where(key, value)
+    .then((results) => results.map(convertFromPersistent))
+    .then((results) => {
+      if (results.length === 1) {
+        return results[0]
+      } else if (results.length === 0) {
+        return null
+      } else {
+        assert(false)
+      }
+    })
+}
+
+function getAccount (name, options) {
+  return getAccountByKey('NAME', name, options)
+}
+
+function getAccountByFingerprint (fingerprint, options) {
+  return getAccountByKey('FINGERPRINT', fingerprint, options)
+}
+
+function * updateAccount (account, options) {
+  return getTransaction(options)(TABLE_NAME)
+    .update(convertToPersistent(account))
+    .where('NAME', account.name)
+}
+
+function * insertAccount (account, options) {
+  return getTransaction(options)
+    .insert(convertToPersistent(account))
+    .into(TABLE_NAME)
 }
 
-function * getAccount (name, options) {
-  return yield Account.findByName(name, options)
+function * insertAccounts (accounts, options) {
```

Comment:
I had been following the previous convention of calling this method `bulkInsert`. I'm fine with this name, but we should be consistent in both places. Incidentally, do you think we need "Account" in the method names, since these are all in the `db/accounts` file? Could do `get`, `getAll`, `getByFingerprint`, etc. I don't have a strong preference, but something to think about.